text/0730-semver-for-ts.md

```diff
@@ -0,0 +1,912 @@
+
+[3.5-breakage]: https://github.com/microsoft/TypeScript/issues/33272
+[3.7-emit-change]: https://github.com/microsoft/TypeScript/pull/33470
+
+
+##### Symbols
+
+Changing a symbol is a breaking change when:
+
+-   changing the name of an exported symbol (type or value), since users' existing imports will need to be updated. This is a breaking for value exports (`let`, `const`, `class`, `function`) independent of types, but renaming exported `interface`, `type` alias, or `namespace` declarations is breaking as well.
+
+-   removing an exported symbol, since users' existing imports will stop working. This is a breaking change for value exports (`let`, `const`, `class`, `function`) independent of types, but removing exported `interface`, `type` alias, or `namespace` declarations is breaking as well.
+
+    This includes changing a previously type-and-value export such as `export class` to either—
+    
+    -   a type-only export, since the exported value symbol has been removed:
+
+        ```diff
+        -export class Foo {
+        +class Foo {
+          neato: string;
+        }
+        +
+        +export type { Foo };
+        ```
+    
+    -   a value-only export, since the exported type symbol has been removed:
+
+        ```diff
+        -export class Foo {
+        +class _Foo {
+          neato: string;
+        }
+        +
+        +export let Foo: typeof _Foo;
+        ```
+
+-   changing the kind (*value* vs. *type*) of an exported symbol in any way, since users' imports and own definitions may both be broken, since imports resolve all symbols imported together if they share a name:
+
+    -   Given a *value*-only exported symbol, including `namespace` declarations, adding a *type* export with the same name as the *value* may break users' code: they may have imported the value and safely created a type of the same name. Their existing import will now cause a re-declaration conflict. Note that this is distinct from adding an entirely new type export where there was no type or value export previously, since the user could never accidentally introduce the conflict, and could work around the conflict using the `as` import specifier when introducing the import.
+
+    -   Given a *type*-only exported symbol, including `type`, `interface`, or `export type` for a type or value, adding a *value* export with the same name may break users' code: they may have imported the type and safely created a value of the same name. Their existing import will now cause a re-declaration conflict. Note that this is distinct from adding an entirely new value export where there was no type or value export previously, since the user could never accidentally introduce the conflict, and could work around the conflict using the `as` import specifier when introducing the import.
```

Comment:
No, because "type-only" imports still import it into the value space for `typeof` purposes:
```ts
import type { Component } from "react"
type X = typeof Component // ok
```

Reply:
Made this explicit in 8620856, crediting both of you for the discussion.